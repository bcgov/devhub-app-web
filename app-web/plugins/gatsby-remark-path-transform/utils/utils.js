/*
Copyright 2018 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/

const path = require('path');
const jsdom = require('jsdom'); // eslint-disable-line

const { JSDOM } = jsdom;

/**
 * based on the files location within the repo
 * the relative asset path declared in the file
 * and the repo
 * this will return an absolute location of the asset
 * path
 * @param {String} filePath 
 * @param {String} repoURI 
 * @param {String} relativePath 
 * @returns {String} the absolute version of the relative asset path
 */
const assetRelativePathToAbsolute = (filePath, repoURI, relativePath) => {
    // get directory path from filePath
    try {
        const pathToFile = path.parse(filePath).dir;
        const absoluteAssetPath = repoURI + path.resolve(`${pathToFile}`, relativePath);
        return absoluteAssetPath;
    } catch(e) {
        return relativePath;  
    }
    
};

const isRelativePath = path => /^\.\.?\//.test(path);

/**
 * sifts over transformed markdown content (which should be html by now)
 * and replace any relative asset links in hrefs or src attributes with
 * absolute versions
 * @param {String} filePath 
 * @param {String} repoURI 
 * @param {String} content 
 */
const convertRelativePathsToAbsolute = (filePath, repoURI, content) => {
    // jsdomify content
    const dom = new JSDOM(content);
    const { window: { document } } = dom;
    const anchors = document.getElementsByTagName('a');
    const images = document.getElementsByTagName('img');
    const videos = document.getElementsByTagName('video');

    const assetTags = Array.from(anchors)
        .concat(Array.from(images))
        .concat(Array.from(videos));
    // loop over asset tags and relace hrefs or srcs if they are relative
    assetTags.forEach(assetTag => {
        const attributeType = assetTag.tagName.toLocaleLowerCase() === 'a' ? 'href' : 'src'; 
        // does assetTag have a src or href attribute?
        const assetPath = assetTag.getAttribute(attributeType);
        if(isRelativePath(assetPath)) {
            const absoluteAssetPath = assetRelativePathToAbsolute(filePath, repoURI, assetPath);
            assetTag.setAttribute(attributeType, absoluteAssetPath);
        }
    });
    return document.getElementsByTagName('body')[0].innerHTML;
};

module.exports = {
    assetRelativePathToAbsolute,
    isRelativePath,
    convertRelativePathsToAbsolute,
};