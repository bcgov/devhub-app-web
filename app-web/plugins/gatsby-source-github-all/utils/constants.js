//
// Dev Hub
//
// Copyright Â© 2018 Province of British Columbia
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
// Created by Patrick Simonian on 2018-10-12.
//

// source types that map to fetch routines
const SOURCE_TYPES = {
  GITHUB: 'github',
};

const GRAPHQL_NODE_TYPE = 'DevhubSiphon';

const DEFUALT_IGNORES = [
  'node_modules',
  'CONTRIBUTING.md',
  'CODE_OF_CONDUCT.md',
  'LICENSE',
  'CODE OF CONDUCT.md',
  'openshift',
];
// github rest api v3
const GITHUB_API_ENDPOINT = 'https://api.github.com';

const FILETYPES = {
  yml: 'YAML',
  yaml: 'YAML',
  json: 'JSON',
  md: 'Markdown',
  txt: 'text',
  html: 'HTML',
  HTML: 'HTML',
};

const MEDIATYPES = {
  md: 'text/markdown',
  json: 'application/json',
  yaml: 'text/yaml',
  yml: 'text/yaml',
  txt: 'text/html',
  html: 'text/html',
  HTML: 'text/html',
};

const PROCESSABLE_EXTENSIONS = ['.md', '.yml', '.yaml', '.html'];

const MARKDOWN_FRONTMATTER_SCHEMA = {
  title: {
    type: String,
    required: false,
  },
  description: {
    type: String,
    required: true,
  },
  resourcePath: {
    type: String,
    required: false,
  },
  ignore: {
    type: Boolean,
    required: false,
  },
};

const GITHUB_SOURCE_SCHEMA = {
  url: {
    type: String,
    required: true,
  },
  owner: {
    type: String,
    required: true,
  },
  repo: {
    type: String,
    required: true,
  },
};

module.exports = {
  GRAPHQL_NODE_TYPE,
  MARKDOWN_FRONTMATTER_SCHEMA,
  PROCESSABLE_EXTENSIONS,
  DEFUALT_IGNORES,
  GITHUB_API_ENDPOINT,
  GITHUB_SOURCE_SCHEMA,
  FILETYPES,
  MEDIATYPES,
  SOURCE_TYPES,
};
