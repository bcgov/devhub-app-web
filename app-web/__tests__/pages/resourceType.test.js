import React from 'react';
import { render, cleanup } from '@testing-library/react';
import queryString from 'query-string';
import { ThemeProvider } from 'emotion-theming';
import theme from '../../theme';
import { ResourceType } from '../../src/templates/resourceType';
import { SIPHON_NODES, TOPICS, GITHUB_RAW_NODES, EVENTS, JOURNEYS } from '../../__fixtures__/nodes';
import { SELECT_RESOURCES_GROUPED_BY_TYPE } from '../../__fixtures__/selector-fixtures';
import { useSearch } from '../../src/utils/hooks';

import { TEST_IDS as NO_RESOURCE_TEST_IDS } from '../../src/components/UI/NoResources/NoResources';
import { SEARCH as SEARCH_MESSAGES } from '../../src/messages';
import DEFAULT_FILTERS, { FILTER_QUERY_PARAM } from '../../src/constants/filterGroups';
import { TEST_IDS as FILTER_TEST_IDS } from '../../src/components/Filters/FilterGroup/FilterGroup';
import { RESOURCE_TYPES } from '../../src/constants/ui';
import { removeOtherResourceTypeResults } from '../../src/utils/helpers';
// this adds custom jest matchers from jest-dom
jest.mock('query-string');
// mock out layout
jest.mock('../../src/hoc/Layout.js', () => ({ children }) => children);
// mock out search hook
jest.mock('../../src/utils/hooks.js');

describe('Resource Type Template Page', () => {
  // mock out non redux selectors
  jest.doMock('../../src/utils/selectors.js', () => ({
    selectResourcesGroupedByType: jest.fn(() => SELECT_RESOURCES_GROUPED_BY_TYPE),
  }));
  // when you use graphql to load data into the component
  // all edges are an object of { node: [graphql object]}
  // the topics fixture is the true data without this extra object field
  // so we map it to resemble what graphql would do when passing the data attribute into
  // this component
  const nodes = SIPHON_NODES.map(c => ({ node: c }));
  const topics = TOPICS.map(c => ({ node: c }));
  const journeys = JOURNEYS.map(c => ({ node: c }));
  const githubRaw = GITHUB_RAW_NODES.map(c => ({ node: c }));
  const props = {
    data: {
      allDevhubSiphon: {
        edges: nodes,
      },
      allDevhubTopic: {
        edges: topics,
      },
      allGithubRaw: {
        edges: githubRaw,
      },
      allTopicRegistryJson: {
        edges: topics,
      },
      allJourneyRegistryJson: {
        edges: journeys,
      },
    },
    location: {
      search: '',
    },
    pageContext: {
      resourceTypeConst: 'DOCUMENTATION',
      resourceType: 'Documentation',
    },
  };

  afterEach(() => {
    cleanup();
    useSearch.mockClear();
  });

  test('it matches snapshot, when there are no resources for the given reosource type the no resource component shows ', () => {
    queryString.parse.mockReturnValue({});
    const { container, rerender, queryByTestId } = render(
      <ThemeProvider theme={theme}>
        <ResourceType {...props} />
      </ThemeProvider>,
    );
    expect(container.firstChild).toMatchSnapshot();

    expect(queryByTestId(NO_RESOURCE_TEST_IDS.container)).not.toBeInTheDocument();

    const newSiphonNodes = SIPHON_NODES.filter(
      node => node.fields.resourceType !== RESOURCE_TYPES.DOCUMENTATION,
    ).map(node => ({ node }));

    const newprops = {
      ...props,
      data: {
        ...props.data,
        allDevhubSiphon: {
          edges: newSiphonNodes,
        },
        allGithubRaw: {
          edges: [], // simulating zero github raw nodes
        },
      },
    };

    rerender(
      <ThemeProvider theme={theme}>
        <ResourceType {...newprops} />
      </ThemeProvider>,
    );

    expect(queryByTestId(NO_RESOURCE_TEST_IDS.container)).toBeInTheDocument();
  });

  test('when searching, if there are results the number of cards should reduce', () => {
    queryString.parse.mockReturnValue({});
    useSearch.mockReturnValue([]);
    const { rerender, queryAllByText } = render(
      <ThemeProvider theme={theme}>
        <ResourceType {...props} />
      </ThemeProvider>,
    );
    const startingNumCards = queryAllByText(RESOURCE_TYPES.DOCUMENTATION).length;

    queryString.parse.mockReturnValue({ q: 'foo' });

    useSearch.mockReturnValue([{ id: SIPHON_NODES[0].id }]);
    rerender(
      <ThemeProvider theme={theme}>
        <ResourceType {...props} />
      </ThemeProvider>,
    );

    const endingNumCards = queryAllByText(RESOURCE_TYPES.DOCUMENTATION).length;

    expect(startingNumCards).toBeGreaterThan(endingNumCards);
  });

  test('when searching, if there are zero results the alert box should show', () => {
    queryString.parse.mockReturnValue({ q: 'foo' });
    useSearch.mockReturnValue([]);
    const { queryByText } = render(
      <ThemeProvider theme={theme}>
        <ResourceType {...props} />
      </ThemeProvider>,
    );
    const Alert = queryByText(SEARCH_MESSAGES.results.empty.defaultMessage);

    expect(Alert).toBeInTheDocument();
  });

  test('Cards are not duplicating after multiple searches', () => {
    queryString.parse.mockReturnValue({ q: 'foo' });
    useSearch.mockReturnValue([{ id: SIPHON_NODES[0].id }]);
    const { rerender, queryAllByText } = render(
      <ThemeProvider theme={theme}>
        <ResourceType {...props} />
      </ThemeProvider>,
    );

    const startingNumCards = queryAllByText(RESOURCE_TYPES.DOCUMENTATION).length;

    queryString.parse.mockReturnValue({ q: 'foo' });
    useSearch.mockReturnValue([{ id: SIPHON_NODES[0].id }]);
    rerender(
      <ThemeProvider theme={theme}>
        <ResourceType {...props} />
      </ThemeProvider>,
    );

    const endingNumCards = queryAllByText(RESOURCE_TYPES.DOCUMENTATION).length;

    expect(startingNumCards).toEqual(endingNumCards);
  });

  test('when there are filters, the resource count should reduce', () => {
    const firstFilterKey = DEFAULT_FILTERS[0].key;
    // mock out query string returning a 'filter'
    queryString.parse.mockReturnValue({});

    useSearch.mockReturnValue([]);
    const { queryAllByText, rerender } = render(
      <ThemeProvider theme={theme}>
        <ResourceType {...props} />
      </ThemeProvider>,
    );

    const startingNumCards = queryAllByText(RESOURCE_TYPES.DOCUMENTATION).length;

    queryString.parse.mockReturnValue({ [FILTER_QUERY_PARAM]: firstFilterKey });

    rerender(
      <ThemeProvider theme={theme}>
        <ResourceType {...props} />
      </ThemeProvider>,
    );

    const endingNumCards = queryAllByText(RESOURCE_TYPES.DOCUMENTATION).length;

    expect(startingNumCards).toBeGreaterThan(endingNumCards);
  });

  test('when given the set of resource (by type) and a given filter is not applicable to the resources (not filterable), it should be disabled ', () => {
    queryString.parse.mockReturnValue({});
    useSearch.mockReturnValue([]);
    // modify the siphon nodes so that none of them reference the first filter, we should expect then that that filter
    // should be disabled, but the remaining filters should be enabled
    const firstFilter = DEFAULT_FILTERS[0];
    const secondFilter = DEFAULT_FILTERS[1];
    const thirdFilter = DEFAULT_FILTERS[2];

    // these filters filter by attributes.personas which is an array value inside of the nodes
    const siphonNodesNotFilterableByFirstFilter = SIPHON_NODES.map(node => ({
      ...node,
      fields: {
        ...node.fields,
        personas: node.attributes.personas
          .filter(p => p !== firstFilter.value)
          .concat(secondFilter.value),
      },
    }));

    const githubRawNodesNotFilterableByFirstFilter = GITHUB_RAW_NODES.map(node => ({
      ...node,
      fields: {
        ...nodes.fields,
        personas: node.fields.personas
          .filter(p => p !== firstFilter.value)
          .concat(thirdFilter.value),
      },
    }));

    const { queryAllByTestId } = render(
      <ThemeProvider theme={theme}>
        <ResourceType
          {...props}
          data={{
            ...props.data,
            allDevhubSiphon: {
              edges: siphonNodesNotFilterableByFirstFilter.map(node => ({ node })),
            },
            allGithubRaw: {
              edges: githubRawNodesNotFilterableByFirstFilter.map(node => ({ node })),
            },
          }}
        />
      </ThemeProvider>,
    );

    const FirstCheckBox = queryAllByTestId(`${FILTER_TEST_IDS.checkbox}-${firstFilter.key}`);
    const SecondCheckbox = queryAllByTestId(`${FILTER_TEST_IDS.checkbox}-${secondFilter.key}`);
    const ThirdCheckbox = queryAllByTestId(`${FILTER_TEST_IDS.checkbox}-${thirdFilter.key}`);

    expect(FirstCheckBox[0]).toHaveAttribute('disabled');
    expect(SecondCheckbox[0]).not.toHaveAttribute('disabled');
    expect(ThirdCheckbox[0]).not.toHaveAttribute('disabled');
  });

  test('when given the set of resource (by type) where only 1 filter is filterable, it should be disabled because it is redundant ', () => {
    queryString.parse.mockReturnValue({});
    // modify the siphon nodes so that none of them reference the first filter, we should expect then that that filter
    // should be disabled, but the remaining filters should be enabled
    const firstFilter = DEFAULT_FILTERS[0];
    const secondFilter = DEFAULT_FILTERS[1];
    const thirdFilter = DEFAULT_FILTERS[2];

    // these filters filter by attributes.personas which is an array value inside of the nodes
    const siphonNodesNotFilterableByFirstFilter = SIPHON_NODES.map(node => ({
      ...node,
      fields: {
        ...node.fields,
        personas: [firstFilter.value], // set in so all nodes have the same filter value
      },
      attributes: {
        ...node.attributes,
      },
    }));

    const { queryAllByTestId } = render(
      <ThemeProvider theme={theme}>
        <ResourceType
          {...props}
          data={{
            ...props.data,
            allDevhubSiphon: {
              edges: siphonNodesNotFilterableByFirstFilter.map(node => ({ node })),
            },
          }}
        />
      </ThemeProvider>,
    );
    // technically the first checkbox is filterable, but because it is the only filter available
    // it is disabled since it is redundant
    const FirstCheckBox = queryAllByTestId(`${FILTER_TEST_IDS.checkbox}-${firstFilter.key}`);
    const SecondCheckbox = queryAllByTestId(`${FILTER_TEST_IDS.checkbox}-${secondFilter.key}`);
    const ThirdCheckbox = queryAllByTestId(`${FILTER_TEST_IDS.checkbox}-${thirdFilter.key}`);

    expect(FirstCheckBox[0]).toHaveAttribute('disabled');
    expect(SecondCheckbox[0]).toHaveAttribute('disabled');
    expect(ThirdCheckbox[0]).toHaveAttribute('disabled');
  });

  test('Search results from other resource types are removed from the results', () => {
    //Initial results contain various different resource types
    const initialResults = SIPHON_NODES.concat(EVENTS);
    //since we say the given resource type for this page is Components, events should be filtered out of the results
    const filteredResults = removeOtherResourceTypeResults(
      initialResults,
      'Components',
      SIPHON_NODES.concat(EVENTS),
    );
    expect(initialResults.length).toBeGreaterThan(filteredResults.length);
  });
});
