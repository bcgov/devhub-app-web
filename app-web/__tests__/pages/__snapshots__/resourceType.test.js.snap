// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resource Type Template Page it matches snapshot, when there are no resources for the given reosource type the no resource component shows  1`] = `
.emotion-66 {
  font-size: 15px;
  max-width: 1300px;
  margin: 0;
  padding: 10px 10px;
}

@media screen and (min-width:932px) {
  .emotion-66 {
    padding: 10px 65px;
  }
}

.emotion-2 {
  margin-bottom: 10px;
  color: #494949;
}

.emotion-2 h1 {
  margin-bottom: 5px;
}

.emotion-0 {
  font-size: 1em;
  max-width: 565px;
  margin-bottom: 0;
  padding: 3px;
  line-height: 1.5em;
}

.emotion-64 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-4 {
  -webkit-flex: 0 0 200px;
  -ms-flex: 0 0 200px;
  flex: 0 0 200px;
  padding: 5px;
  margin-right: 15px;
  border: 2px solid #ccc;
  min-height: 250px;
  display: none;
}

@media (min-width:576px) {
  .emotion-4 {
    display: block;
  }
}

.emotion-62 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  width: 100%;
  max-width: 1080px;
  margin-bottom: 20px;
}

@media (min-width:576px) {
  .emotion-62 {
    margin-bottom: 15px;
    max-width: 810px;
  }
}

@media (min-width:768px) {
  .emotion-62 {
    margin-bottom: 15px;
    max-width: 1080px;
  }
}

.emotion-10 {
  padding: 0 5px;
  margin-left: 5px;
}

@media (min-width:576px) {
  .emotion-10 {
    max-width: 415px;
  }
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-8 > input {
  margin: 0 4px;
  padding: 4px 5px;
  -webkit-flex-basis: 250px;
  -ms-flex-preferred-size: 250px;
  flex-basis: 250px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
}

.emotion-8 > button {
  padding: 4px 15px;
  margin: 4px;
}

.emotion-6 {
  color: #fff;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 1px solid transparent;
  padding: 0.375rem 0.75rem;
  font-size: 1em;
  line-height: 1.5;
  border-radius: 0.25rem;
  cursor: pointer;
  font-weight: 400;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  display: inline-block;
  background-color: #003366;
}

.emotion-12 {
  background-color: #fafafa;
  border: 1px solid #ccc;
  padding: 10px;
  margin: 10px 5px 5px;
  display: block;
}

@media (min-width:576px) {
  .emotion-12 {
    display: none;
  }
}

.emotion-60 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media (min-width:810px) {
  .emotion-60 {
    -webkit-box-pack: start;
    -webkit-justify-content: start;
    -ms-flex-pack: start;
    justify-content: start;
  }
}

.emotion-35 {
  margin: 0 10px;
  -webkit-flex: 0 0 250px;
  -ms-flex: 0 0 250px;
  flex: 0 0 250px;
}

.emotion-33 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #1A5A96;
  font-weight: 400;
  text-transform: capitalize;
  padding: 0 4px;
  margin: 0 2px;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: initial;
}

.emotion-33:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: initial;
  cursor: pointer;
}

.emotion-30 {
  width: 265px;
  height: 238.5px;
  margin: 10px 0;
  border: 1px solid #ccc;
  border-top: transparent;
  border-radius: 2px;
  box-sizing: border-box;
  -webkit-transition: -webkit-transform 0.25s ease-out;
  -webkit-transition: transform 0.25s ease-out;
  transition: transform 0.25s ease-out;
  box-shadow: 0 1px 2px 1px #00000026;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  overflow: hidden;
}

.emotion-30:hover {
  box-shadow: 0 2px 2px 1px #00000026;
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.emotion-14 {
  background-color: #246BD1;
  -webkit-flex: 0 0 10px;
  -ms-flex: 0 0 10px;
  flex: 0 0 10px;
  min-height: 10px;
}

.emotion-28 {
  padding: 6px 10px;
  height: 100%;
  overflow: auto;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
}

.emotion-18 {
  color: #444;
  font-size: 15px;
  margin-bottom: 0;
  margin-top: 0;
  font-weight: 400;
}

.emotion-16 {
  color: #246BD1;
}

.emotion-20 {
  font-size: 18px;
  color: #494949;
  line-height: 1.3;
  margin-bottom: 0;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  text-transform: capitalize;
  word-break: break-word;
}

.emotion-22 {
  line-height: 1.4;
  font-size: 14px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  margin-bottom: 0;
  margin-top: 4px;
  text-transform: none;
}

.emotion-26 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  overflow: hidden;
  padding-top: 10px;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-24 {
  width: 100%;
  height: auto;
  object-fit: scale-down;
  max-height: 100%;
  margin-bottom: 0;
}

<main
  class="emotion-66 emotion-67"
  role="main"
>
  <div
    class="emotion-2 emotion-3"
  >
    <h1>
      Documentation
    </h1>
    <h4
      class="emotion-0 emotion-1"
    >
      Explore documentation to common Government Entities, Protocols and Compliances
    </h4>
  </div>
  <div
    class="emotion-64 emotion-65"
  >
    <div
      class="emotion-4 emotion-5"
    >
      <div>
        <div
          class="form-group"
        >
          <label
            class=""
            data-testid="filter-group-title-For"
            for="For"
            style="font-weight: 600; color: rgb(73, 73, 73); font-size: 14px; margin-bottom: 0px;"
          >
            For
          </label>
          <div
            class="Checkbox custom-checkbox custom-control"
          >
            <input
              aria-label="Filter resources by this value"
              class="custom-control-input"
              data-testid="filter-group-checkbox-designer"
              id="For0"
              type="checkbox"
            />
            <label
              class="custom-control-label"
              for="For0"
            >
              Designers
            </label>
          </div>
          <div
            class="Checkbox custom-checkbox custom-control"
          >
            <input
              aria-label="Filter resources by this value"
              class="custom-control-input"
              data-testid="filter-group-checkbox-developer"
              id="For1"
              type="checkbox"
            />
            <label
              class="custom-control-label"
              for="For1"
            >
              Developers
            </label>
          </div>
          <div
            class="Checkbox custom-checkbox custom-control"
          >
            <input
              aria-label="Filter resources by this value"
              class="custom-control-input"
              data-testid="filter-group-checkbox-product-owner"
              id="For2"
              type="checkbox"
            />
            <label
              class="custom-control-label"
              for="For2"
            >
              Product Owners
            </label>
          </div>
        </div>
      </div>
    </div>
    <div
      class="emotion-62 emotion-63"
    >
      <div
        class="emotion-10 emotion-11"
      >
        <div
          class="emotion-8 emotion-9"
        >
          <input
            aria-label="Search input field"
            class="Input custom-text"
            data-testid="searchbar-input"
            id="dh-search-input"
            maxlength="40"
            type="text"
            value=""
          />
          <button
            aria-label="Click to Search"
            class="emotion-6 emotion-7"
            color="blue"
            data-testid="searchbar-button"
          >
            Search
          </button>
        </div>
        
      </div>
      <div
        class="emotion-12 emotion-13"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-filter fa-w-16 "
          data-icon="filter"
          data-prefix="fas"
          focusable="false"
          role="img"
          style="color: rgb(0, 34, 102); margin-right: 5px;"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M487.976 0H24.028C2.71 0-8.047 25.866 7.058 40.971L192 225.941V432c0 7.831 3.821 15.17 10.237 19.662l80 55.98C298.02 518.69 320 507.493 320 487.98V225.941l184.947-184.97C520.021 25.896 509.338 0 487.976 0z"
            fill="currentColor"
          />
        </svg>
         
        <span>
          Filters
        </span>
      </div>
      <div
        class="emotion-60 emotion-61"
      >
        <div
          class="emotion-35 emotion-36"
        >
          <a
            class="emotion-32 emotion-33 emotion-34"
            data-testid="ui-gatsby-link"
            href="/DevHub-Node-1"
          >
            <article
              class="emotion-30 emotion-31"
            >
              <div
                class="emotion-14 emotion-15"
                color="Documentation"
              />
              <div
                class="emotion-28 emotion-29"
              >
                <div
                  style="margin-bottom: 6px;"
                >
                  <h3
                    class="emotion-18 emotion-19"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-file fa-w-12  emotion-16 emotion-17"
                      data-icon="file"
                      data-prefix="fas"
                      data-testid="resource-type-icon"
                      focusable="false"
                      role="img"
                      type="Documentation"
                      viewBox="0 0 384 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm160-14.1v6.1H256V0h6.1c6.4 0 12.5 2.5 17 7l97.9 98c4.5 4.5 7 10.6 7 16.9z"
                        fill="currentColor"
                      />
                    </svg>
                    <span
                      style="margin: 4px;"
                    >
                      Documentation
                    </span>
                    <small>
                      <svg
                        aria-hidden="true"
                        class="svg-inline--fa fa-external-link-alt fa-w-18 "
                        data-icon="external-link-alt"
                        data-prefix="fas"
                        focusable="false"
                        role="img"
                        viewBox="0 0 576 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M576 24v127.984c0 21.461-25.96 31.98-40.971 16.971l-35.707-35.709-243.523 243.523c-9.373 9.373-24.568 9.373-33.941 0l-22.627-22.627c-9.373-9.373-9.373-24.569 0-33.941L442.756 76.676l-35.703-35.705C391.982 25.9 402.656 0 424.024 0H552c13.255 0 24 10.745 24 24zM407.029 270.794l-16 16A23.999 23.999 0 0 0 384 303.765V448H64V128h264a24.003 24.003 0 0 0 16.97-7.029l16-16C376.089 89.851 365.381 64 344 64H48C21.49 64 0 85.49 0 112v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V287.764c0-21.382-25.852-32.09-40.971-16.97z"
                          fill="currentColor"
                        />
                      </svg>
                    </small>
                  </h3>
                </div>
                <div
                  class="emotion-20 emotion-21"
                >
                  About
                </div>
                <p
                  class="emotion-22 emotion-23"
                >
                  What the Design System is and how it works.
                </p>
                <div
                  class="emotion-26 emotion-27"
                >
                  <img
                    alt="About"
                    class="emotion-24 emotion-25"
                    src="https://github.com/bcgov/design-system/blob/master/components/about/about.md?raw=true"
                  />
                </div>
              </div>
            </article>
          </a>
        </div>
        <div
          class="emotion-35 emotion-36"
        >
          <a
            class="emotion-32 emotion-33 emotion-34"
            data-testid="ui-gatsby-link"
            href="/GitHub-Raw-1"
          >
            <article
              class="emotion-30 emotion-31"
            >
              <div
                class="emotion-14 emotion-15"
                color="Documentation"
              />
              <div
                class="emotion-28 emotion-29"
              >
                <div
                  style="margin-bottom: 6px;"
                >
                  <h3
                    class="emotion-18 emotion-19"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-file fa-w-12  emotion-16 emotion-17"
                      data-icon="file"
                      data-prefix="fas"
                      data-testid="resource-type-icon"
                      focusable="false"
                      role="img"
                      type="Documentation"
                      viewBox="0 0 384 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm160-14.1v6.1H256V0h6.1c6.4 0 12.5 2.5 17 7l97.9 98c4.5 4.5 7 10.6 7 16.9z"
                        fill="currentColor"
                      />
                    </svg>
                    <span
                      style="margin: 4px;"
                    >
                      Documentation
                    </span>
                    <small>
                      <svg
                        aria-hidden="true"
                        class="svg-inline--fa fa-external-link-alt fa-w-18 "
                        data-icon="external-link-alt"
                        data-prefix="fas"
                        focusable="false"
                        role="img"
                        viewBox="0 0 576 512"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M576 24v127.984c0 21.461-25.96 31.98-40.971 16.971l-35.707-35.709-243.523 243.523c-9.373 9.373-24.568 9.373-33.941 0l-22.627-22.627c-9.373-9.373-9.373-24.569 0-33.941L442.756 76.676l-35.703-35.705C391.982 25.9 402.656 0 424.024 0H552c13.255 0 24 10.745 24 24zM407.029 270.794l-16 16A23.999 23.999 0 0 0 384 303.765V448H64V128h264a24.003 24.003 0 0 0 16.97-7.029l16-16C376.089 89.851 365.381 64 344 64H48C21.49 64 0 85.49 0 112v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V287.764c0-21.382-25.852-32.09-40.971-16.97z"
                          fill="currentColor"
                        />
                      </svg>
                    </small>
                  </h3>
                </div>
                <div
                  class="emotion-20 emotion-21"
                >
                  Introduction To Github and Gov
                </div>
                <p
                  class="emotion-22 emotion-23"
                >
                  Everything you need to know about working with Github in the BC Government.
                </p>
                <div
                  class="emotion-26 emotion-27"
                >
                  <img
                    alt="Introduction To Github and Gov"
                    class="emotion-24 emotion-25"
                    src="../images/octokat.png"
                  />
                </div>
              </div>
            </article>
          </a>
        </div>
      </div>
    </div>
  </div>
</main>
`;
